cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreManager.java

```diff
@@ -57,12 +58,17 @@
  */
 public class DatastoreManager {
 
+    // Static map to manage DatastoreManager instances (one per directory path)
+    private static final ConcurrentMap<File, DatastoreManager> datastoreManagers = new
+            ConcurrentHashMap<File, DatastoreManager>();
+
     private final static String LOG_TAG = "DatastoreManager";
     private final static Logger logger = Logger.getLogger(DatastoreManager.class.getCanonicalName());
 
     private final String path;
 
-    private final Map<String, Datastore> openedDatastores = Collections.synchronizedMap(new HashMap<String, Datastore>());
+    /* This map should only be accessed inside synchronized(openDatastores) blocks */
```

Comment:
What's the tradeoff between 'manually' synchronising and using `Collections.synchronizedMap`? I gather the code previously did both which is obviously wrong.